src/source/agc.rs

```diff
@@ -74,9 +72,11 @@ impl CircularBuffer {
     #[inline]
     fn push(&mut self, value: f32) -> f32 {
         let old_value = self.buffer[self.index];
+        // Update the sum by first subtracting the old value and then adding the new value; this is more accurate.
+        self.sum = self.sum - old_value + value;
         self.buffer[self.index] = value;
-        self.sum += value - old_value;
-        self.index = (self.index + 1) % self.buffer.len();
+        // Use bitwise AND for efficient index wrapping since RMS_WINDOW_SIZE is a power of two.
```

Comment:
pretty sure the compiler will think of that :)
if your unsure benchmark it?

Reply:
Did benchmark it, it's a little bit better with this, plus this makes sure the compiler is doing the optimization.

Reply:
oh you did, well now im impressed by your thoroughness and disappointed in the compiler :sweat_smile: